R/fit.R

```diff
@@ -391,16 +391,21 @@ mmrm <- function(formula,
   if (control$method %in% c("Kenward-Roger", "Kenward-Roger-Linear") && !reml) {
     stop("Kenward-Roger only works for REML")
   }
+  covariance <- h_reconcile_cov_struct(formula, covariance)
+  formula_parts <- h_mmrm_tmb_formula_parts(formula, covariance)
 
-  attr(data, which = "dataname") <- toString(match.call()$data)
+  if (!missing(data)) {
+    attr(data, which = "dataname") <- toString(match.call()$data)
+  } else {
+    # na.action set to na.pass to allow data to be full; will be futher trimmed later
+    data <- model.frame(formula_parts$full_formula, na.action = "na.pass")
```

Comment:
create data will use full data (because the length of the variable should match from the beginning)